frontend/packages/pipelines-plugin/src/components/repository/list-page/RepositoryRow.tsx

```diff
@@ -0,0 +1,115 @@
+import * as React from 'react';
+import { TableRow, TableData, RowFunction } from '@console/internal/components/factory';
+import { Kebab, ResourceKebab, ResourceLink, Timestamp } from '@console/internal/components/utils';
+import { referenceFor, referenceForModel } from '@console/internal/module/k8s';
+import { PipelineRunModel, RepositoryModel } from '../../../models';
+import { pipelineRunFilterReducer } from '../../../utils/pipeline-filter-reducer';
+import { pipelineRunDuration } from '../../../utils/pipeline-utils';
+import LinkedPipelineRunTaskStatus from '../../pipelineruns/status/LinkedPipelineRunTaskStatus';
+import PipelineRunStatus from '../../pipelineruns/status/PipelineRunStatus';
+import { useGetLastPLR } from '../repository-hooks';
+import { RepositoryKind } from '../types';
+import { repositoriesTableColumnClasses } from './RepositoryHeader';
+
+type RepositoryPLRProps = {
+  pipelineRunsName: string[];
+  namespace: string;
+};
+
+export const PLRStatus: React.FC<RepositoryPLRProps> = ({ pipelineRunsName, namespace }) => {
+  const latestRun = useGetLastPLR(pipelineRunsName, namespace);
+  return (
+    <PipelineRunStatus
+      status={pipelineRunFilterReducer(latestRun)}
+      title={pipelineRunFilterReducer(latestRun)}
+      pipelineRun={latestRun}
+    />
+  );
+};
+
+export const PLRRuntime: React.FC<RepositoryPLRProps> = ({ pipelineRunsName, namespace }) => {
+  const latestRun = useGetLastPLR(pipelineRunsName, namespace);
+  return <Timestamp timestamp={latestRun?.status.startTime} />;
+};
+
+export const PLRDuration: React.FC<RepositoryPLRProps> = ({ pipelineRunsName, namespace }) => {
+  const latestRun = useGetLastPLR(pipelineRunsName, namespace);
+  return <>{pipelineRunDuration(latestRun)}</>;
+};
+
+export const PLRLink: React.FC<RepositoryPLRProps> = ({ pipelineRunsName, namespace }) => {
+  const latestRun = useGetLastPLR(pipelineRunsName, namespace);
+  return (
+    <ResourceLink
+      kind={referenceForModel(PipelineRunModel)}
+      name={latestRun?.metadata.name}
+      namespace={namespace}
+    />
+  );
+};
+
+export const PLRTask: React.FC<RepositoryPLRProps> = ({ pipelineRunsName, namespace }) => {
+  const latestRun = useGetLastPLR(pipelineRunsName, namespace);
+  return latestRun && <LinkedPipelineRunTaskStatus pipelineRun={latestRun} />;
+};
```

Comment:
Why did you break these out to their own components? I'm not against it, but please make them proper component files if you are doing this for bite-sized chunks. One component per file is our convention.

If you only did this for testing purposes, could you help me understand why?

Reply:
I did this so, that I can use the hook. But I found another way to do it. So removed these components.